imagej/imagej.py

```diff
@@ -286,19 +294,159 @@ def run_script(self, language, script, args=None):
 
         def to_java(self, data):
             """
-            Converts the data into a java equivalent.  For numpy arrays, the java image points to the python array
+            Converts the data into a java equivalent.  For numpy arrays, the java image points to the python array.
+
+            In addition to the scyjava types, we allow ndarray-like and xarray-like variables
             """
             if self._is_memoryarraylike(data):
                 return imglyb.to_imglib(data)
+            if self._is_xarraylike(data):
+                return self.to_dataset(data)
             return to_java(data)
 
         def to_dataset(self, data):
+            """Converts the data into an ImageJ dataset"""
+            if self._is_xarraylike(data):
+                return self._xarray_to_dataset(data)
+            if self._is_arraylike(data):
+                return self._numpy_to_dataset(data)
+            if scyjava.isjava(data):
+                return self._java_to_dataset(data)
+
+            raise TypeError(f'Type not supported: {type(data)}')
+
+        def _numpy_to_dataset(self, data):
+            rai = imglyb.to_imglib(data)
+            return self._java_to_dataset(rai)
+
+        def _xarray_to_dataset(self, xarr):
+            """
+            Converts a xarray dataarray with specified dim order to an image
+            :param xarr: Pass an xarray dataarray and turn into a dataset.
+            :return: The dataset
+            """
+            dataset = self._numpy_to_dataset(xarr.values)
+            axes = self._assign_axes(xarr)
+            dataset.setAxes(axes)
+
+            # Currently, we have no handling for nonlinear axes, but I thought it should warn instead of fail.
+            if not self._axis_is_linear(xarr.coords):
+                warnings.warn("Not all axes are linear.  The nonlinear axes are not mapped correctly.")
+            
+            self._assign_dataset_metadata(dataset, xarr.attrs)
+
+            return dataset
+        
+        def _assign_axes(self, xarr):
+            """
+            Obtain xarray axes names, origin, and scale and convert into ImageJ Axis; currently supports DefaultLinearAxis.
+            :param xarr: xarray that holds the units
+            :return: A list of ImageJ Axis with the specified origin and scale
+            """
+            axes = ['']*len(xarr.dims)
+
+            for axis in xarr.dims:
+                origin = self._get_origin(xarr.coords[axis])
+                scale = self._get_scale(xarr.coords[axis])
+
+                axisStr = self._pydim_to_ijdim(axis)
+
+                ax_type = Axes.get(axisStr)
+                ax_num = self._get_axis_num(xarr, axis)
+                if scale is None:
+                    java_axis = DefaultLinearAxis(ax_type)
+                else:
+                    java_axis = DefaultLinearAxis(ax_type, numpy.double(scale), numpy.double(origin))
```

Comment:
Is there a difference between `numpy.double` and Python's built-in `float`?

Reply:
Yes, numpy.double works on a list/multiple values.  That's how it is used here.